libtelemetry/src/full/java/com/mapbox/android/telemetry/MapboxTelemetry.java

```diff
@@ -479,10 +479,12 @@ public Thread newThread(Runnable runnable) {
   }
 
   @SuppressWarnings("WeakerAccess")
-  public void setBaseUrl(String eventsHost) {
-    if (isValidUrl(eventsHost)) {
+  public synchronized boolean setBaseUrl(String eventsHost) {
+    if (isValidUrl(eventsHost) && checkNetworkAndParameters()) {
```

Comment:
do you need to check for network here?

Reply:
`TelemetryClient` is created inside that check. It's a pipeline used by `MapboxTelemetry` class to init `TelemetryClient`

Reply:
`checkNetworkAndParameters()` checks for both network connection and init as required, so if you need the network check then this is fine, if the intent is to check and init `TelemetryClient` then you could  `checkRequiredParameters(..)`

Reply:
we discussed it internally and decided to check network connection as well.